pod: add gauge for runtimeclass handler

egernst opened the pull request:
**What this PR does / why we need it**:

This adds a metric for observability of the RuntimeClass feature:
kube_pod_runtimeclass_handler

Using this metrics allows end-users to quickly verify that the RuntimeClass feature is being utilized and is healthy.

When combined with the kube_pod_status_phase, end-users can use tools like grafana to quickly assess how pods running with a particular runtimeclass are behaving compared to other runtimeclasses and the default runtime. 

**Which issue(s) this PR fixes**

Fixes: https://github.com/kubernetes/kube-state-metrics/issues/1274

egernst commented:
/cc @SergeyKanzhelev @tallclair

SergeyKanzhelev commented:
I have limited context here. From looking at the code around it looks consistent and seems to be taking the proper value for the dimension. As next step maybe we can add supported runtime handlers to the [kube_node_info](https://github.com/kubernetes/kube-state-metrics/blob/2e1ff1f529ea72fc857d1e595a85cfaab57a2c80/docs/node-metrics.md#user-content-node-metrics:~:text=kube_node_info)

/lgtm

egernst commented:
> I have limited context here. From looking at the code around it looks consistent and seems to be taking the proper value for the dimension. As next step maybe we can add supported runtime handlers to the [kube_node_info](https://github.com/kubernetes/kube-state-metrics/blob/2e1ff1f529ea72fc857d1e595a85cfaab57a2c80/docs/node-metrics.md#user-content-node-metrics:~:text=kube_node_info)
> 
> /lgtm

Interesting. I'm not sure there's an easy way for the node to know in a clean way what runtime handlers are available. RuntimeClass *can* optionally specify the nodeSelect, but that isn't always necessarily used.

lilic commented on internal/store/pod.go:
```diff
@@ -1511,6 +1511,27 @@ var (
+			}),
+		),
+		*generator.NewFamilyGenerator(
+			"kube_pod_runtimeclass_handler",
```
Can we add a unit test for this, thanks! :)

lilic replied:
As per best practices the name should be `kube_pod_runtimeclass_handler_info`, can you adjust that as well, thanks!

lilic commented:
> As next step maybe we can add supported runtime handlers to the kube_node_info

What would be the use of that metric?

lilic commented:
To answer the issue question: If you manage to address the comment before our next release (which will be tomorrow most likely) we can merge this into 2.0.0-alpha.2 cut, otherwise it will have to wait until 2.0 is shipped as we are going into code freeze after that for 2.0.

egernst commented:
Thanks for the quick feedback @lilic.  I updated the commit to include unit test (whoops sorry!) and to add `_info` to the metric name.

I also changed the metric name from runtimeclass_handler to runtimeclass_name.

lilic reviewed (approved):
/lgtm

Thanks!
